drape_frontend/watch/point.h

```diff
@@ -247,7 +247,7 @@ inline int segment(std::vector<point_base<T> > const & v, double length, double
   }
   if (out_length)
     *out_length = length;
-  return segment_num;
+  return static_cast<int>(segment_num);
```

Comment:
I suggest you to return size_t instead of int. And when you need to return -1, just return number of segments.